### PR TITLE
ui: use StatementsPage from admin-ui-components

### PR DESCRIPTION
**Pull request by koorosh**
Depends on https://github.com/cockroachdb/admin-ui-components/pull/9
Depends on https://github.com/cockroachdb/yarn-vendored/pull/27
Depends on https://github.com/cockroachdb/yarn-vendored/pull/28

`StatementsPage` component is replaced with previously extracted version of StatementsPage in `admin-ui-components`.
The new version (extracted) of `StatementsPage` has several changes that required to make the following adjustments:
- `ActivateStatementDiagnostics` modal isn't connected components anymore so dispatching actions are pulled up to connected StatementsPage wrapper.
- Previously, analytics functions (which track user interaction) were
called directly from React components that made this functionality tightly coupled.
Now, sagas used to watch actions related to user interaction and then call async logic in sagas.
-- components expose only callback functions which are related to the
the logic component is responsible;
-- connected component is responsible to dispatch actions with proper
payload to be properly then handled by sagas.

### CODE REVIEW

**Comment by support-dpulls** [comment]
Hey there, Walid from [Dpulls](https://www.dpulls.com/) here!
We noticed that you have created a PR that depends on another one on a different repository. However, **Dpulls** is currently not installed on the repository `cockroachdb/admin-ui-components` which will cause the dependency to not be resolved in case the PR `cockroachdb/admin-ui-components#9` gets merged (because Dpulls won't receive the event).
So we suggest you to also install **Dpulls** on `cockroachdb/admin-ui-components` using this link https://github.com/apps/dpulls/installations/new.
There is an open issue to address this kind of problem https://github.com/dpulls/backlog/issues/7. Basically in the future, if a dependency repo is not installed, the **Dpulls**  bot will add a comment on the PR mentioning that and the status check won't be created.
Let me know if you need more information. Thanks!
cc @koorosh @jordanlewis

**Review by dhartunian — CHANGES_REQUESTED** [review]
@koorosh the Statements Page storybook entry is currently broken. Can you take a look?

File: pkg/ui/src/redux/analyticsActions.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export const TRACK_STATEMENTS_PAGINATION = "cockroachui/analytics/TRACK_STATEMENTS_PAGINATION";
+
+export function trackStatementsSearchAction(searchResults: number): PayloadAction<number> {
+  return {
+    type: TRACK_STATEMENTS_SEARCH,
+    payload: searchResults,
+  };
+}
+
+export function trackStatementsPaginationAction(pageNum: number): PayloadAction<number> {
```
[the remark this answers is not in the document]
Comment (koorosh) [reply pkg/ui/src/redux/analyticsActions.ts:23]: Thanks, David! indeed it has to be restored!

**Comment by koorosh** [comment]
> @koorosh the Statements Page storybook entry is currently broken. Can you take a look?

@dhartunian the entire StatementsPage [storybook](https://github.com/cockroachdb/admin-ui-components/blob/master/src/statementsPage/statementsPage.stories.tsx) is moved to `admin-ui-component` which consumes mocked data.
I suppose to remove this duplicate storybook here until we have a way to connect storybooks with the real server.
What do you think?

**Comment by dhartunian** [comment]
> > @koorosh the Statements Page storybook entry is currently broken. Can you take a look?
> 
> @dhartunian the entire StatementsPage [storybook](https://github.com/cockroachdb/admin-ui-components/blob/master/src/statementsPage/statementsPage.stories.tsx) is moved to `admin-ui-component` which consumes mocked data.
> I suppose to remove this duplicate storybook here until we have a way to connect storybooks with the real server.
> What do you think?

Yes we can remove. Sounds good 👍

**Review by dhartunian — APPROVED** [review]

**Pull request merged**